### PR TITLE
Changes when filtering is done in an default textinput column

### PR DESCRIPTION
**Pull request by JockeCK**
Now filtering starts when user presses enter, blur occurs or 1 second passes without additional user text input.

Unsure if this can make it to the circuit release since alpha has started. Leaving project blank

Refs https://github.com/Neovici/cosmoz-omnitable/issues/60

### CODE REVIEW

File: cosmoz-omnitable-column.js [hunk 1]
```diff
@@ -61,6 +61,7 @@ class OmnitableColumn extends columnMixin(PolymerElement) {
 
 	resetFilter() {
 		this.filter = null;
+		this.filterCandidate = null;
```
Comment (Wurper) [inline cosmoz-omnitable-column.js]:
```suggestion
		this.inputValue = null;
```

**Comment by JockeCK** [comment]
Seems I am failing 15 tests or so. will take an look

File: cosmoz-omnitable-column.js [hunk 2]
```diff
@@ -61,6 +73,33 @@ class OmnitableColumn extends columnMixin(PolymerElement) {
...
+		}
+	}
+
+	_inputValueValueChanged() {
+		if (this.inputValue === '') {
+			this.filter = this.inputValue;
+			return;
+		}
+		this._debouncer = Debouncer.debounce(this._debouncer,
+			timeOut.after(1000),
```
Comment (nomego) [inline cosmoz-omnitable-column.js]: Do we want this as a configurable property?
Or maybe at least a constant in the file?

File: cosmoz-omnitable-column.js [hunk 3]
```diff
@@ -61,6 +73,33 @@ class OmnitableColumn extends columnMixin(PolymerElement) {
...
+		case 13: // Enter
+			event.preventDefault();
+			this.filter = this.inputValue;
+			break;
+		}
+	}
+
+	_inputValueValueChanged() {
+		if (this.inputValue === '') {
+			this.filter = this.inputValue;
```
Comment (nomego) [inline cosmoz-omnitable-column.js]: Maybe add a test step and assert for this?

File: cosmoz-omnitable-column.js (hunk 2 above)
Comment (JockeCK) [reply cosmoz-omnitable-column.js]: See latest commit: Like this?

File: cosmoz-omnitable-column.js (hunk 3 above)
Comment (JockeCK) [reply cosmoz-omnitable-column.js]: See latest commit: Like this?

**Comment by JockeCK** [comment]
@cristinecula I am getting an failed test in column-number.html:
out of range values are limited on blur ‣
Error: expected 2 to equal 5
  Context.<anonymous> at column-number.html:144

It is the flush(I think) I do when I added two new tests in basic.html. 

Any idea what the best way is to do here?

File: test/basic.html [hunk 4]
```diff
@@ -119,6 +122,27 @@
 				assert.equal(column.sortOn, 'sortOnValuePath');
 			});
 
+			test('changing inputValue updates filter eventually', () => {
+				const column = omnitable.columns.find(col => {
```
Comment (nomego) [inline test/basic.html]:
```suggestion
				const column = omnitable.columns.find(col => col.name === 'columnWithFilter');
```

File: test/basic.html [hunk 5]
```diff
@@ -58,6 +60,7 @@
 	<script type="module">
 		import { setupOmnitableFixture } from './helpers/utils';
 		import { generateTableDemoData } from '../demo/table-demo-helper';
+		import { flush as flush$0 } from '@polymer/polymer/lib/utils/flush';
```
Comment (nomego) [inline test/basic.html]: Do we need to rename flush? Why ?

**Comment by cristinecula** [comment]
@JockeCK When working on these tests, I encountered a weird issue regarding blur events, which may cause your failing test: chrome ignores blur/focus events when the browser window is not focused. This means that if you run the tests interactively, you must always keep the window focused, otherwise you will have intermittent test failures. This is not an issue when running the tests in headless browsers. Can you confirm that this is not the cause of the test failure?

**Comment by JockeCK** [comment]
> @JockeCK When working on these tests, I encountered a weird issue regarding blur events, which may cause your failing test: chrome ignores blur/focus events when the browser window is not focused. This means that if you run the tests interactively, you must always keep the window focused, otherwise you will have intermittent test failures. This is not an issue when running the tests in headless browsers. Can you confirm that this is not the cause of the test failure?

@cristinecula You are right! Kept the window up and focused and the test did not fail. Thank you!

**Comment by cristinecula** [comment]
@JockeCK great! I think you should add a paragraph to `test/index.html` to mention this fact.

File: test/basic.html (hunk 4 above)
Comment (JockeCK) [reply test/basic.html]: Refactored, and all the rest in basic.html

File: test/basic.html (hunk 5 above)
Comment (JockeCK) [reply test/basic.html]: No need, removed renaming.

File: test/index.html [hunk 6]
```diff
@@ -8,6 +8,8 @@
 	<script src="/components/wct-mocha/wct-mocha.js"></script>
 </head>
 <body>
+	<!-- If running tests interactively, some tests may fail if the window is not focused during the tests.
```
Comment (cristinecula) [inline test/index.html]: @JockeCK Maybe it's a good idea to make this text visible, so it's seen when tests are run interactively.

Comment (JockeCK) [reply test/index.html]: yes

**Pull request merged**